enterprise/core-edge/src/main/java/org/neo4j/coreedge/raft/roles/Candidate.java

```diff
@@ -115,11 +115,9 @@ else if ( res.term() < ctx.term() || !res.voteGranted() )
 
                 if ( req.term() > ctx.term() )
                 {
-                    outcome.setNextTerm( req.term() );
                     outcome.getVotesForMe().clear();
-
                     outcome.setNextRole( FOLLOWER );
-                    outcome.addOutgoingMessage( new RaftMessages.Directed<>( ctx.myself(), req ) );
+                    Voting.handleVoteRequest( ctx, outcome, req );
```

Comment:
Same for this one actually. Handling vote requests is always done in the context of Follower.